src/signed-xml.ts

```diff
@@ -1020,36 +1303,243 @@ export class SignedXml {
     this.signatureNode = signatureDoc;
     const signedInfoNodes = utils.findChildren(this.signatureNode, "SignedInfo");
     if (signedInfoNodes.length === 0) {
-      const err3 = new Error("could not find SignedInfo element in the message");
-      if (!callback) {
-        throw err3;
+      throw new Error("could not find SignedInfo element in the message");
+    }
+    const signedInfoNode = signedInfoNodes[0];
+
+    return {
+      doc,
+      prefix,
+      signatureDoc,
+      signedInfoNode,
+    };
+  }
+
+  /**
+   * Compute the signature of the given XML (using the already defined settings).
+   *
+   * @param xml The XML to compute the signature for.
+   * @param callback A callback function to handle the signature computation asynchronously.
+   * @returns void
+   * @throws TypeError If the xml can not be parsed.
+   */
+  computeSignature(xml: string): void;
```

Comment:
There is something off with this diff, maybe whitespace? Can you see what might be going on as this is very difficult to review when the diff is so misaligned and unchanged lines appear changed.